pkg/config/config.go

```diff
@@ -23,92 +26,52 @@ type (
+		PersistRedisCluster map[string]*Persist       `json:"persist_redis_cluster,omitempty" yaml:"persist_redis_cluster,omitempty" toml:"persist_redis_cluster,omitempty"`
+		PubSub              map[string]*PubSub        `json:"pubsub,omitempty" yaml:"pubsub,omitempty" toml:"pubsub,omitempty"`
+		Config              map[string]*Config        `json:"config,omitempty" yaml:"config,omitempty" toml:"config,omitempty"`
 	}
 
-	Config struct {
-		Type        string      `json:"type" yaml:"type" toml:"type"`
-		InfraParams InfraParams `json:"pulumi_params,omitempty" yaml:"pulumi_params,omitempty" toml:"pulumi_params,omitempty"`
-	}
-
-	ExecutionUnit struct {
-		Type                 string            `json:"type" yaml:"type" toml:"type"`
-		NetworkPlacement     string            `json:"network_placement,omitempty" yaml:"network_placement,omitempty" toml:"network_placement,omitempty"`
-		EnvironmentVariables map[string]string `json:"environment_variables,omitempty" yaml:"environment_variables,omitempty" toml:"environment_variables,omitempty"`
-		HelmChartOptions     *HelmChartOptions `json:"helm_chart_options,omitempty" yaml:"helm_chart_options,omitempty" toml:"helm_chart_options,omitempty"`
-		InfraParams          InfraParams       `json:"pulumi_params,omitempty" yaml:"pulumi_params,omitempty" toml:"pulumi_params,omitempty"`
+	ContentDeliveryNetwork struct {
+		Id string `json:"id,omitempty" yaml:"id,omitempty" toml:"id,omitempty"`
 	}
 
-	// A HelmChartOptions represents configuration for execution units attempting to generate helm charts
-	HelmChartOptions struct {
-		Directory   string   `json:"directory,omitempty" yaml:"directory,omitempty" toml:"directory,omitempty"` // Directory signals the directory which will contain the helm chart outputs
-		Install     bool     `json:"install,omitempty" yaml:"install,omitempty" toml:"install,omitempty"`
-		ValuesFiles []string `json:"values_files,omitempty" yaml:"values_files,omitempty" toml:"values_files,omitempty"`
-	}
+	/** InfraParams are passed as-is to the generated IaC
+	*   It follows a generic map and is later converted to the Kind specific object
+	 */
 
-	PubSub struct {
-		Type        string      `json:"type" yaml:"type" toml:"type"`
-		InfraParams InfraParams `json:"pulumi_params,omitempty" yaml:"pulumi_params,omitempty" toml:"pulumi_params,omitempty"`
-	}
+	InfraParams map[string]interface{}
 
-	StaticUnit struct {
-		Type                   string                 `json:"type" yaml:"type" toml:"type"`
-		InfraParams            InfraParams            `json:"pulumi_params,omitempty" yaml:"pulumi_params,omitempty" toml:"pulumi_params,omitempty"`
-		ContentDeliveryNetwork ContentDeliveryNetwork `json:"content_delivery_network,omitempty" yaml:"content_delivery_network,omitempty" toml:"content_delivery_network,omitempty"`
+	KindParams interface {
+		Merge(cfg interface{}) InfraParams
```

Comment:
so this can actually be removed all together, leftover from an attempt to do the kind params as interfaces, but after our discussion this morning its not needed. will remove it